### ISSUE TITLE
java.lang.RuntimeException: Unrecoverable error while evaluating node, Caused by: java.lang.NullPointerException

### ISSUE THREAD

**Issue by jfirebaugh** [bazelbuild/bazel#15304]
### Description of the bug:

Bazel crashes with a NullPointerException in com.google.devtools.build.lib.rules.cpp when trying to build a target.

```
FATAL: bazel crashed due to an internal error. Printing stack trace:
java.lang.RuntimeException: Unrecoverable error while evaluating node 'ConfiguredTargetKey{label=@cxx.rs//:core, config=BuildConfigurationValue.Key[94ad3ff1c3b6daff2fe5835bbe476df6d2aa03ed110865074d10167bed18fa61]}' (requested by nodes 'ConfiguredTargetKey{label=//:hello-world-cc, config=BuildConfigurationValue.Key[94ad3ff1c3b6daff2fe5835bbe476df6d2aa03ed110865074d10167bed18fa61]}')
	at com.google.devtools.build.skyframe.AbstractParallelEvaluator$Evaluate.run(AbstractParallelEvaluator.java:674)
	at com.google.devtools.build.lib.concurrent.AbstractQueueVisitor$WrappedRunnable.run(AbstractQueueVisitor.java:382)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1426)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Caused by: java.lang.NullPointerException
	at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:880)
	at com.google.devtools.build.lib.rules.cpp.CcCommon.<init>(CcCommon.java:173)
	at com.google.devtools.build.lib.rules.cpp.CcLibrary.init(CcLibrary.java:132)
	at com.google.devtools.build.lib.rules.cpp.CcLibrary.create(CcLibrary.java:95)
	at com.google.devtools.build.lib.rules.cpp.CcLibrary.create(CcLibrary.java:67)
	at com.google.devtools.build.lib.analysis.ConfiguredTargetFactory.createRule(ConfiguredTargetFactory.java:364)
	at com.google.devtools.build.lib.analysis.ConfiguredTargetFactory.createConfiguredTarget(ConfiguredTargetFactory.java:188)
	at com.google.devtools.build.lib.skyframe.SkyframeBuildView.createConfiguredTarget(SkyframeBuildView.java:1108)
	at com.google.devtools.build.lib.skyframe.ConfiguredTargetFunction.createConfiguredTarget(ConfiguredTargetFunction.java:980)
	at com.google.devtools.build.lib.skyframe.ConfiguredTargetFunction.compute(ConfiguredTargetFunction.java:368)
	at com.google.devtools.build.skyframe.AbstractParallelEvaluator$Evaluate.run(AbstractParallelEvaluator.java:590)
	... 7 more
```

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

* Check out the minimized repro: https://github.com/jfirebaugh/emsdk_rules_rust_bug.
* Run `bazel build :hello-world-wasm`.

### Which operating system are you running Bazel on?

macOS

### What is the output of `bazel info release`?

release 5.0.0-homebrew

### What's the output of `git remote get-url origin; git rev-parse master; git rev-parse HEAD` ?

```text
git@github.com:jfirebaugh/emsdk_rules_rust_bug.git
dc9aadf6f7014aff53db191a0705016f5361e001
dc9aadf6f7014aff53db191a0705016f5361e001
```

### Any other information, logs, or outputs that you want to share?

This target attempts to combine rules_cc, rules_rust, and emsdk in order to build, via emscripten, a WASM target whose sources are a mix of C++ and Rust. In order to do so, I've patched rules_rust and emsdk. The patches are included in the minimized repro, and involve fixing https://github.com/emscripten-core/emsdk/issues/984 and then getting rules_rust and emsdk to agree on the same set of platform constraints for their toolchains.

**Comment by jfirebaugh** [comment]
I figured out that the NullPointerException is triggered by the registration by rules_rust of a "dummy_cc_wasm32_toolchain":

https://github.com/bazelbuild/rules_rust/blob/cd44b3670a0b492eccbe48c20027fd3651cd51e3/rust/private/dummy_cc_toolchain/BUILD.bazel#L5-L13

In a project which wants to combine C++ and Rust code into a single wasm executable, I believe the assumption that this code is making ("we know we'll never look [the cc toolchain] up") is false.

**Comment by kcking** [comment]
I ran into a similar issue in trying to build a native rust server that depends on rust compiled to wasm. I get the stack trace from above when `--incompatible_enable_cc_toolchain_resolution=1` is set. I was able to fix this issue by patching rules_rust's dummy cc toolchain to have the fields expected by bazel: https://github.com/kcking/rules_rust/commit/a20fd7cf8ae19dbb37a69107df9dc447e2d5e3c0

For @jfirebaugh, I think even if this NPE is fixed your underlying problem is that you want to use a valid cc toolchain alongside rust in a single wasm32 target. I would suggest preventing rules_rust from [registering the dummy toolchain](https://github.com/bazelbuild/rules_rust/blob/078c6908fc32c168b58e72cc3884dd8e30419e3a/rust/repositories.bzl#L611) at all so that you can provide your own valid toolchain. I couldn't find a way to do this besides using a local fork and commenting out that line, but I'm guessing rules_rust would accept a patch to add a flag if it unblocks you.

**Comment by oquenchil** [comment]
This should be obsolete with the Starlark implementation of cc_library which will be released in 6.0.
